src/app/home/home.component.ts

```diff
@@ -8,6 +10,15 @@ import { UserService } from '../services/user.service';
   styleUrls: ['./home.component.css'],
 })
 export class HomeComponent implements OnInit {
-  ngOnInit(): void {}
-  constructor() {}
+  data: Ad[] = [];
+  ad?: Ad;
+  user?: User | undefined;
+  constructor(private adsService: AdsService, userService: UserService) { }
+
+  ngOnInit(): void {
+    this.adsService.getAds().subscribe(result => {
+
```

Comment:
`
this.adsService.getAds().subscribe(result => {
 // for each ad in result.data, if the active === true, then
// push it to data
})
`